### DISCUSSION TITLE
Result of running pathfindR is "NAs introduced by coercion"

### DISCUSSION THREAD

**Discussion by obearap** [egeulgen/pathfindR#100] [Q&A] [answered]
Hi

Since the weekend I've been getting an "NAs introduced by coercion" error message when i run the pathfindR package. An example of some of my data is

```
library ("pathfindR") 

Gene.symbol <- c("ACAA2", "ACADVL", "ACAT1", "ACOT9", "ACOX1", "ADH5",  "AKR1A1")
logFC <- c("3.3", "3.9", "1.5", "1.7", "2.4", "1.9", "1.7")
adj.P.Val <- c("0.02", "0.03", "0.02", "0.007", "0.01", "0.03", "0.03")
PSEA.DXB11 <- data.frame(Gene.symbol, logFC, adj.P.Val)
```

PathfindR is executed with

`PSEA_results.DXB11 <- run_pathfindR(PSEA.DXB11, output_dir = ("C:/PETER PROJECT/3g. Mass Spec Processing (Comparison of K1 v DXB11)/Results/PathfindR/DXB11 p Values/1. Attached")) `

which results in

`NAs introduced by coercion
`
I'm running version 1.6.2. and rebooted my laptop to no effect.

Thanks

**Comment by egeulgen** [comment]
Can you turn logFC and adj.P.Val into numeric vectors? I think the issue is that they’re both character vectors.
Let me know if that helps,
Best,
-E

**Reply by obearap** [reply]
Hi. Turned those columns into numeric vectors as follows:

PSEA.DXB11$logFC <- as.numeric(PSEA.DXB11$logFC)
PSEA.DXB11$adj.P.Val <- as.numeric(PSEA.DXB11$adj.P.Val)
str(PSEA.DXB11) 

but that hasn't solved the problem. 

If it helps this code ran successfully last week so the issue only arose this weekend.

Thanks

**Reply by egeulgen** [reply]
In your local settings, is the decimal place a dot or a comma?

can you also share the entire console output? This should be a warning and not an error if I’m not mistaken. With such a small number of genes, it’s not surprising that you get an empty result(I’m assuming the result is an empty data frame)

**Reply by obearap** [reply]
With regards to the number of genes, what I've posted here is a subset of the 100 genes in my dataset. The issue arises when I run the entire dataset. I don't mind sending that full dataset to you but would prefer not to publish it on a discussion forum as the data hasn't been published yet.

The decimal place is a dot in all cases. Below is the output of the str command for the full dataset.

`str(PSEA.DXB11)`

'data.frame':	100 obs. of  3 variables:
 $ Gene.symbol: chr  "ACAA2" "ACADVL" "ACAT1" "ACOT9" ...
 $ logFC      : num  3.33 3.93 1.52 1.7 2.38 ...
 $ adj.P.Val  : num  3.49e-07 1.34e-05 2.47e-04 1.99e-04 4.93e-05

Re the message I actually don't get anything in the console other than a repeating loop of the phrase "NAs introduced by coercion" which can only be stopped by terminating R studio.

**Reply by egeulgen** [reply]
If you don't mind, can you save the data frame as an RDS and email me at egeulgen [at] gmail.com?
I'll try to replicate the issue

**Comment by obearap** [comment, marked as answer]
This can be closed. It turned out to be a local machine issue and uninstalling/reinstalling R studio has solved the problem
